packages/backend/src/submission/service.ts

```diff
@@ -94,30 +117,59 @@ async function getAllSubmissions(repository: SubmissionRepository) {
 }
 
 async function addSubmission(
-  repository: SubmissionRepository,
+  submissionRepository: SubmissionRepository,
+  entityRepository: EntityRepository,
+  didResolver: DidResolver,
   payload: Record<string, unknown>,
 ): Promise<Submission> {
   const submissionDto = SubmissionDto.parse(payload)
-
-  if (!(await repository.findInvitationById(submissionDto.invitationId))) {
-    throw new Error('Invitation not found')
-  }
   if (
-    await repository.findSubmissionByInvitationId(submissionDto.invitationId)
+    !(await submissionRepository.findInvitationById(submissionDto.invitationId))
   ) {
-    throw new Error('Submission already completed')
+    throw new Error('Invitation not found')
   }
-  if (await repository.findSubmissionByDid(submissionDto.did)) {
-    throw new Error('Submission with the same DID already exists')
+  for (const did of submissionDto.dids) {
+    const didDocument = await didResolver.resolveDid(did)
+    if (!didDocument) {
+      throw new Error(`DID '${did}' is not resolvable`)
+    }
+    const existingEntity = await entityRepository.findByDid(did)
+    if (
+      existingEntity?.invitationId &&
+      existingEntity.invitationId !== submissionDto.invitationId
+    ) {
+      throw new Error(
+        `An entity associated with a different invitation already contains the DID '${did}'`,
+      )
+    }
   }
-  const submission = {
-    ...submissionDto,
-    id: uuidv4(),
-    createdAt: new Date().toISOString(),
-    updatedAt: new Date().toISOString(),
+
+  const pendingSubmission =
+    await submissionRepository.findPendingSubmissionByInvitationId(
+      submissionDto.invitationId,
+    )
+
+  let submission: Submission
+  if (pendingSubmission) {
+    submission = {
+      ...pendingSubmission,
+      ...submissionDto,
+      updatedAt: new Date().toISOString(),
+    }
+    await submissionRepository.updateSubmission(submission)
```

Comment:
But it would be a different submission, and different submission ID? Only `pending` state submissions are updated.

Reply:
I think it would be the same. Admin opens the submission with ID 123 in state `pending` and while looking at it, before he approves it, the submitter sends another submission, then we do an update of ID 123, but the admin sees just the previous of submission with ID 123.

Reply:
Then perhaps we should implement some sort of webhook feature where we listen for changes on a particular submission while the admin is viewing it? Or we can verify the data is the same when the admin attempts to approve / decline, and if not, do not alert the admin that changes have been made since latest review.

Reply:
Removed ability to update a submission